garak/generators/rest.py

```diff
@@ -193,31 +195,44 @@ def _call_model(
             "timeout": self.request_timeout,
         }
         resp = self.http_function(self.uri, **req_kArgs)
+
+        if resp.status_code in self.skip_codes:
+            logging.debug(
+                "REST skip prompt: %s - %s, uri: %s",
+                resp.status_code,
+                resp.reason,
+                self.uri,
+            )
+            return [None]
+
         if resp.status_code in self.ratelimit_codes:
-            raise RateLimitHit(f"Rate limited: {resp.status_code} - {resp.reason}, uri: {self.uri}")
+            raise RateLimitHit(
+                f"Rate limited: {resp.status_code} - {resp.reason}, uri: {self.uri}"
+            )
 
-        elif str(resp.status_code)[0] == "3":
+        if str(resp.status_code)[0] == "3":
             raise NotImplementedError(
                 f"REST URI redirection: {resp.status_code} - {resp.reason}, uri: {self.uri}"
             )
 
-        elif str(resp.status_code)[0] == "4":
+        if str(resp.status_code)[0] == "4":
             raise ConnectionError(
                 f"REST URI client error: {resp.status_code} - {resp.reason}, uri: {self.uri}"
             )
 
-        elif str(resp.status_code)[0] == "5":
+        if str(resp.status_code)[0] == "5":
             error_msg = f"REST URI server error: {resp.status_code} - {resp.reason}, uri: {self.uri}"
             if self.retry_5xx:
                 raise IOError(error_msg)
-            else:
-                raise ConnectionError(error_msg)
+            raise ConnectionError(error_msg)
 
         if not self.response_json:
             return [str(resp.text)]
 
         response_object = json.loads(resp.content)
 
+        response = [None] * generations_this_call
```

Comment:
Since `self.supports_multiple_generations` is set to `False` in the in the constructor should we opt for consistency [with the skip case](https://github.com/leondz/garak/pull/999/files#diff-ff96cd9116f6456af6b492c39bc684effd08ab800cdf6f33e9d1bffead1ac309R206) and and avoid the multiplier here? 
```suggestion
        response = [None]
```

If/when `self.supports_multiple_generations` can be supported the entire method will need to be rechecked to ensure all `return` statements conform to required `generations_this_call`.

Reply:
OK